spec/models/step_by_step_page_spec.rb

```diff
@@ -134,6 +137,68 @@
     }.to raise_error(ActiveRecord::RecordNotUnique)
   end
 
+  describe 'broken link checking' do
+    let(:step_by_step_with_step) { create(:step_by_step_page) }
+
+    it 'checks if a link check was requested and there are no broken links' do
+      step = create(:step, step_by_step_page: step_by_step_with_step)
+
+      link_checker_api_get_batch(
+        id: 1,
+        links: [
+          {
+            "uri": "https://www.gov.uk/404",
+            "status": "broken",
+            "checked": "2017-04-12T16:30:39Z",
+            "errors": [
+              "Received 404 response from the server."
+            ],
+            "warnings": [],
+            "problem_summary": "404 error (page not found)",
+            "suggested_fix": ""
+          }
+        ]
+      )
+
+      create(:link_report, batch_id: 1, step_id: step.id)
+
+      expect(step_by_step_with_step.broken_links?).to be true
+      expect(step_by_step_with_step.link_report_requested?).to be true
+    end
+
+    it "checks if a link check was requested and there is at least one broken link" do
```

Comment:
same as above

Reply:
Good spot! Thanks! Will change and fix-up.